mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -588,7 +588,7 @@ private[ml] object DefaultParamsReader {
    */
   def loadMetadata(path: String, sc: SparkContext, expectedClassName: String = ""): Metadata = {
     val metadataPath = new Path(path, "metadata").toString
-    val spark = SparkSession.getActiveSession.get
+    val spark = SparkSession.builder().sparkContext(sc).getOrCreate()
```

Comment:
Hi, @WeichenXu123 , @HyukjinKwon , @zhengruifeng .

This sounds like a regression of 
- https://github.com/apache/spark/pull/47366

If we cannot get an existing one, I believe we should not create SparkSession here.

Can we recover the existing code like the following?
> sc.parallelize(Seq(metadataJson), 1).saveAsTextFile(metadataPath)

Reply:
It will not be a regression. This is Spark ML which is DataFrame-based MLlib by definition. Therefore we should always have default session running. Active session is specified to threads, so it might not exist. Alternatively we could use `SparkSession.getDefaultSession`.